packages/core/orchestration/src/joiner/remote-joiner.ts

```diff
@@ -440,9 +442,18 @@ export class RemoteJoiner {
   }> {
     const { expand, pkField, ids, relationship, options } = params
 
-    let uniqueIds = Array.isArray(ids) ? ids : ids ? [ids] : undefined
+    const isIdsUsingOperatorMap =
+      isObject(ids) && Object.keys(ids).some((key) => !!FilterOperatorMap[key])
 
-    if (uniqueIds) {
+    let uniqueIds = isIdsUsingOperatorMap
+      ? ids
+      : Array.isArray(ids)
+      ? ids
+      : ids
+      ? [ids]
+      : undefined
```

Comment:
Do we have any additional conditional here? I am not sure, but looks like we can write the following. Please double check :)

```ts
let uniqueIds = isIdsUsingOperatorMap
      ? ids
      : Array.isArray(ids)
      ? ids
      : [ids]
```

Reply:
we want to wrap into an array if it is not null or undefined, that is why there is the extra check before wrapping in array or setting it to undefined. But I can clean it up differently